src/Ast/PhpClass.php

```diff
@@ -98,7 +109,14 @@ public function withMembers(Vector $members): PhpClass
     public function toCode(): string
     {
         $extends = is_null($this->extends) ? '' : " extends {$this->extends->toCode()}";
-        $class = $this->final ? 'final class' : 'class';
+
+        $class = 'class';
```

Comment:
nit: let's throw an exception if they're both set
```suggestion
        $class = 'class';
        if ($this->final && $this->abstract) {
            throw new RuntimeException('Cannot use the final modifier on an abstract class');
        }
```

Reply:
+1